source/development/coredump.rst

```diff
@@ -0,0 +1,203 @@
+
+1. **Identify the Target Directory for Core Dumps**:
+    Choose a directory where core dump files will be generated. This directory should have sufficient disk space and appropriate permissions for the process generating core dumps to write to it.
+
+2. **Set the Core Dump Size Limit**:
+    If the current core dump size limit is insufficient, increase it using the ``ulimit`` command. For example, to set the core dump size limit to unlimited:
+
+.. code-block:: console
+
+    ulimit -c unlimited
+..
+
+3. **Enable Core Dump Generation**:
+    Enable core dump generation by setting the ``core_pattern`` sysctl parameter to specify the core dump file pattern and location. For example, to set the core dump file pattern to ``/var/core/core.%e.%p`` (where `%e` represents the executable name and `%p` represents the process ID):
+
+.. code-block:: console
+
+    echo "/var/core/core.%e.%p" > /proc/sys/kernel/core_pattern
+..
+
+4. **Automate Configuration (Optional)**:
+    To preserve these changes across reboots, add the ``ulimit`` command and ``echo`` command setting ``core_pattern`` to a startup script or system initialization script (e.g., ``/etc/rc.local``).
+
+    By following these steps, you can set up core dump generation manually without relying on systemd. However, keep in mind that the process may vary slightly depending on the Linux distribution and version you are using.
+
+
+To disable core dump generation  you can directly adjust system-wide settings and configurations. Here's how you can do it:
+
+1. **Identify the Target Directory for Core Dumps**:
+    If core dumps are currently being generated, identify the directory where they are stored. By default, core dumps may be stored in the current working directory or in the directory specified by the ``core_pattern`` sysctl parameter.
+
+2. **Disable Core Dump Generation**:
+    To disable core dump generation, set the core dump size limit to zero using the ``ulimit`` command:
+
+.. code-block:: console
+
+    ulimit -c 0
+..
+
+3. **Optional: Configure Core Dump Storage Location**:
+    If core dumps were previously being stored, you may want to configure the ``core_pattern`` sysctl parameter to prevent any future core dumps from being generated. For example, you can set it to ``/dev/null`` to discard core dumps:
+
+.. code-block:: console
+
+    echo "/dev/null" > /proc/sys/kernel/core_pattern
+..
+
+4. **Restart Processes if Necessary**:
+If you've changed the ``core_pattern`` parameter, consider restarting relevant processes to ensure that the changes take effect.
+
+By following these steps, you can disable core dump generation without relying on systemd. This approach directly modifies system-wide settings to prevent core dumps from being generated.
+
+Using `systemd`
+^^^^^^^^^^^^^^^
+
+To set up core dump generation using systemd, you can utilize systemd's built-in features for managing core dumps. Here's a step-by-step guide:
+
+1. **Check Core Dump Configuration**:
```

Comment:
Check what? What to look for? Can we include an output example?

Reply:
done